valohai_yaml/objs/step.py

```diff
@@ -44,6 +45,7 @@ def __init__(
         icon: Optional[str] = None,
         category: Optional[str] = None,
         source_path: Optional[str] = None,
+        resources: Optional[OrderedDict] = None,
```

Comment:
I think you missed this? this should also take in `WorkloadResouces` and not a serialized dict 😉

Reply:
That was because of the custom handling of the resources; after fixing that this now works with the proper type as well; will update.

Reply:
Fixed.